File: DoctorChatApi/extract.js
const fs = require('fs');

try {
    const inputFile = process.argv[2];
    const outputFile = process.argv[3];

    const text = fs.readFileSync(inputFile, 'utf8');
    const lines = text.split('\n');
    let inCode = false;
    let code = [];

    for (const line of lines) {
        if (line.trim() === '```csharp') {
            inCode = true;
            continue;
        }
        if (inCode && line.trim() === '```') {
            inCode = false;
            continue;
        }
        if (inCode) {
            code.push(line);
        }
    }

    fs.writeFileSync(outputFile, code.join('\n'));
    console.log(`Done extracting ${code.length} lines.`);
} catch (e) {
    console.error(e);
}

File: DoctorChatApi/extract.py
import sys

def main():
    try:
        with open(sys.argv[1], 'r', encoding='utf-8') as f:
            lines = f.readlines()
            
        code = []
        in_code = False
        for line in lines:
            if line.strip() == '```csharp':
                in_code = True
                continue
            if in_code and line.strip() == '```':
                in_code = False
                continue
            if in_code:
                code.append(line)
                
        with open(sys.argv[2], 'w', encoding='utf-8') as f:
            f.writelines(code)
        print(f"Successfully extracted {len(code)} lines of C# code.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()
